core/pom.xml

```diff
@@ -98,6 +98,23 @@
             </path>
           </annotationProcessorPaths>
         </configuration>
+        <executions>
+          <execution>
+            <id>compile-java9</id>
+            <goals>
+              <goal>compile</goal>
+            </goals>
+            <configuration>
+              <release>9</release>
+              <compileSourceRoots>
+                <compileSourceRoot>${project.basedir}/src/main/java9</compileSourceRoot>
+              </compileSourceRoots>
+              <multiReleaseOutput>true</multiReleaseOutput>
+              <useIncrementalCompilation>false</useIncrementalCompilation>
```

Comment:
I would have hoped that this would cause Maven to recompile everything every time, but it sounds like it actually causes it to [still attempt incremental compilation but in an even worse way](https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#useIncrementalCompilation)? I assume that the motivation is that it saves Maven from checking for other files that could be affected by a change to `module-info.java`, since we know that there should be no such files? I'd probably still leave it out, since it has this misleading name, but I could also see including it but with a comment to clarify its behavior. Is there more than I should be thinking about here?